sys/src/9/amd64/kernel.ld

```diff
@@ -1,78 +1,59 @@
-/* Simple linker script for the ROS kernel.
-   See the GNU ld 'info' manual ("info ld") to learn the syntax. */
-
-/* This script needs to be invoked with -z max-page-size=0x1000.  Otherwise,
- * ld will offset our first section to 1MB within the actual file.  Multiboot
- * requires the header to be in the first two pages. */
+/*
+ * Linker script for Harvey.
+ */
 
-OUTPUT_FORMAT("elf64-x86-64", "elf64-x86-64", "elf64-x86-64")
 ENTRY(_start)
-/* start the kernel at 0x110000. 
- * That way we can use lower ram for critical structures
- */
-KERN_LOAD_ADDR = 0xfffffffff0000000;
 
-SECTIONS
-{
-	/* Entry Linked and loaded at 0x00100000 (includes multiboot) */
-	. = 0x00110000;
+SECTIONS {
+	/*
+	 * start the kernel at 0x0xffff_8000_0020_0000
+	 * This preserves some RAM between 1MiB and the
+	 * start of the kernel for critical structures.
+	 */
+	. = 0xffff800000100000;
 
-	.bootstrap : {
+	PROVIDE(boottext = .);
+	.text.boot : ALIGN(4096) {
 		*(.boottext .bootdata)
+		. = ALIGN(4096);
+		PROVIDE(eboottext = .);
+		. = ALIGN(2097152);
+		PROVIDE(esys = .);
 	}
 
-	/* Link the main kernel for the space after entry + KERN_LOAD_ADDR.  We'll
-	 * still load it adjacent in physical memory */
-	. += KERN_LOAD_ADDR;
-
-	.text : AT(ADDR(.text) - KERN_LOAD_ADDR) {
-		*(.text .stub .text.* .gnu.linkonce.t.*)
+	PROVIDE(text = .);
```

Comment:
don't think text is actually used anywhere?  hard to tell

Reply:
No, but it's useful for debugging: it shows up in a symbol dump, and we can use it to validate that the text segment starts at e.g. KTZERO.